src/components/LoginFormComponent.tsx

```diff
@@ -0,0 +1,92 @@
+import { TextField, Button } from "@material-ui/core";
+import { useCallback, useContext, useState } from "react";
+import styled from "styled-components";
+
+import { TranslationContext } from "../contexts/TranslationContext";
+import { loginAction } from "../reducers/authReducer";
+import { loginRequest, registerRequest } from "../services/authService";
+import { useAppDispatch } from "../store/hooks";
+
+const FormContainer = styled.form`
+  display: grid;
+  gap: 15px;
+  padding: 15px;
+  width: 50%;
+`;
+
+export default function LoginFormComponent() {
+  const dispatch = useAppDispatch();
+  const { loginForm } = useContext(TranslationContext).config;
+
+  const [userName, setUserName] = useState("");
+  const [password, setPassword] = useState("");
+
+  const handleUserNameInput = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+      setUserName(event.target.value);
+    },
+    [userName]
+  );
+
+  const handlePassInput = useCallback(
+    (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+      setPassword(event.target.value);
+    },
+    [password]
+  );
+
+  const loginHandler = useCallback(() => {
+    async function login() {
+      await loginRequest({
+        email: userName,
+        password: password,
+      });
+      dispatch(loginAction({ email: userName, id: 4 }));
+      setUserName("");
+      setPassword("");
+    }
+    login();
+  }, [userName, password]);
+
+  const registerHandler = useCallback(() => {
+    async function register() {
```

Comment:
Use redux for this. 
handle the login state and token, handle the errors, handle async stuff somewhere outside of the component.